lib/puppet/provider/nssdb/certutil.rb

```diff
@@ -0,0 +1,51 @@
+require 'fileutils'
+
+Puppet::Type.type(:nssdb).provide(:certutil) do
+  commands :certutil => 'certutil'
+
+  def create
+    destroy_nssdb
+    create_nssdb_dir
+    generate_nssdb
+  end
+
+  def destroy
+    destroy_nssdb
+  end
+
+  def exists?
+    nssdb_exists?
+  end
+
+  private
+
+  def create_nssdb_dir
+    FileUtils.mkdir_p(resource[:nssdb_dir])
+  end
+
+  def generate_nssdb
+    certutil(
+      '-N',
+      '-d', resource[:nssdb_dir],
+      '-f', resource[:password_file]
+    )
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to generate new NSS database at #{resource[:nssdb_dir]} with password file #{resource[:password_file]}: #{e}")
+    return nil
```

Comment:
Shouldn't this actually raise an error?
```suggestion
    raise Puppet::Error.new("Failed to generate new NSS database at #{resource[:nssdb_dir]} with password file #{resource[:password_file]}: #{e}", e)
```